Wikipedia/assets/codemirror/resources/addon/search/search.js

```diff
@@ -255,16 +283,20 @@
         if (!(match = cursor.findNext())) {
           cursor = getSearchCursor(cm, query);
           state.focusedMatchIndex = 0;
+          state.focusedMatchIndex = -1;
+          state.initialFocusedMatchIndex = -1;
           if (!(match = cursor.findNext()) || (start && cursor.from().line == start.line && cursor.from().ch == start.ch)) {
             focusOnMatch(state); //resets count to 0/0
             return;
           }
         }
-        
+
         state.posFrom = cursor.from(); state.posTo = cursor.to();
-        focusOnMatch(state)
+        focusOnMatch(state);
         if (shouldReplace) {
+          cm.isReplacing = true;
```

Comment:
isReplacing is to get around another bug - intermittently (after looping back around from the end) codemirror would fire a cursorActivity event and our native side which would clear out find/replace (assuming the user tapped the editor).